test/integration/apiserver/apply/apply_test.go

```diff
@@ -244,6 +244,118 @@ func TestNoOpUpdateSameResourceVersion(t *testing.T) {
 	}
 }
 
+// TestNoOpApplyWithEmptyMap
+func TestNoOpApplyWithEmptyMap(t *testing.T) {
+	client, closeFn := setup(t)
+	defer closeFn()
+
+	deploymentName := "no-op"
+	deploymentsResource := "deployments"
+	deploymentBytes := []byte(`{
+		"apiVersion": "apps/v1",
+		"kind": "Deployment",
+		"metadata": {
+			"name": "` + deploymentName + `",
+			"labels": {
+				"app": "nginx"
+			}
+		},
+		"spec": {
+			"replicas": 1,
+			"selector": {
+				"matchLabels": {
+					"app": "nginx"
+				}
+			},
+			"template": {
+				"metadata": {
+					"annotations": {},
+					"labels": {
+						"app": "nginx"
+					}
+				},
+				"spec": {
+					"containers": [{
+						"name": "nginx",
+						"image": "nginx:1.14.2",
+						"ports": [{
+							"containerPort": 80
+						}]
+					}]
+				}
+			}
+		}
+	}`)
+
+	_, err := client.AppsV1().RESTClient().Patch(types.ApplyPatchType).
+		Namespace("default").
+		Param("fieldManager", "apply_test").
+		Resource(deploymentsResource).
+		Name(deploymentName).
+		Body(deploymentBytes).
+		Do(context.TODO()).
```

Comment:
tests should use Background unless the TODO is relevant and there is a plant to switch to a specific context which is not clear ATM.

Reply:
I don't disagree, but this was copy-pasted from adjacent tests, so this is consistent with the surrounding code.

Note that the defacto situation in the code base is:

```
g grep context.TODO | grep _test | wc -l
2581
g grep context.Background | grep _test | wc -l
1073
```

Can we stick with TODO for now and if anyone wants to make this a convention, take it on a separate project?

Reply:
don't mind, up to you as OWNERS.
it's just what golang recommends.